src/main/java/lawrence/app/Lawrence.java

```diff
@@ -0,0 +1,79 @@
+package lawrence.app;
+
+import lawrence.command.Command;
+import lawrence.database.TaskFileManager;
+import lawrence.parser.CommandParser;
+import lawrence.task.Task;
+import lawrence.task.TaskList;
+import lawrence.ui.UserInterface;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Scanner;
+
+/**
+ * The Lawrence chatbot that is able to keep track of tasks and their
+ * respective completion statuses.
+ * <p>
+ * The tasks are saved automatically after every action.
+ * This ensures that all changes are preserved and will be available
+ * the next time the program is run.
+ * </p>
+ * @author Kok Bo Chang
+ */
+public class Lawrence {
+    private static final String NAME = "Lawrence";
+    private static final Path SAVE_LOCATION = Paths.get(".", "data", "tasks.txt");
+
+    private TaskFileManager manager;
+    private TaskList tasks;
+    private UserInterface ui;
+
+    /**
+     * Default constructor.
+     */
+    public Lawrence() {
+        ui = new UserInterface(NAME);
+        manager = new TaskFileManager(SAVE_LOCATION);
+        try {
+            Task[] existingTasks = manager.readTasksFromFile();
+            tasks = new TaskList(existingTasks);
+        } catch (IOException e) {
+            // initialise with no tasks instead
+            tasks = new TaskList(new Task[0]);
+        }
+    }
+
+    /**
+     * Initialises and runs the Lawrence chatbot.
+     *
+     * @param args optional startup arguments
+     */
+    public static void main(String[] args) {
+        Lawrence primeMinister = new Lawrence();
+        primeMinister.run();
+    }
+
+    /**
+     * Runs the chatbot to start listening for user commands
+     * in the console.
+     */
+    public void run() {
+        ui.greetUser();
+
+        Scanner sc = new Scanner(System.in);
+        String userInput;
+        boolean shouldContinue = true;
```

Comment:
Just a nit: consider using a different variable names like isExit

Reply:
`isExit` was used in the example code snippet for the `A-MoreOOP` flag, but I wanted to avoid the double negative that comes with the `while` loop condition check.

To illustrate what I mean:
```
while(!isExit) { // double negative
    // do something
}
```
is a double negative and will take a bit longer for the reader to process the intention of the `while` loop, while
```
while(shouldContinue) {
    // do something
}
```
is much easier to process and understand (in my opinion).

As it stands, using `should` in a Boolean variable is allowing as per the [coding standard](https://se-education.org/guides/conventions/java/index.html#naming). I will keep this unchanged for now, but will keep this in mind and modify things accordingly as the project evolves.